File: frontend/types/reservation.ts
export type ReservationItem = {
  id: string;
  supermarket_product_id: string;
  quantity: number;
  reserved_price: string;
  currency: string;
  product_id: string | null;
  product_name: string | null;
  product_image_url: string | null;
  supermarket_id: string | null;
  supermarket_name: string | null;
  expiration_date: string | null;
};

export type Reservation = {
  id: string;
  status: string;
  items: ReservationItem[];
  created_at: string;
  updated_at: string;
};

File: backend/app/schemas/reservation.py
from datetime import date, datetime
from decimal import Decimal
from typing import Literal
from uuid import UUID

from pydantic import BaseModel


ReservationStatus = Literal["active", "cancelled", "completed", "expired", "inactive"]


class ReservationItemOut(BaseModel):
    id: UUID
    supermarket_product_id: UUID
    quantity: int
    reserved_price: Decimal
    currency: str

    product_id: UUID | None = None
    product_name: str | None = None
    product_image_url: str | None = None

    supermarket_id: UUID | None = None
    supermarket_name: str | None = None
    expiration_date: date | None = None


class ReservationOut(BaseModel):
    id: UUID
    status: str
    items: list[ReservationItemOut]
    created_at: datetime
    updated_at: datetime
